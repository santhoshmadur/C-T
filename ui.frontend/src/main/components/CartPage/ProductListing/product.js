/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React from 'react';
import { FormattedMessage, useIntl } from 'react-intl';
import { gql } from '@apollo/client';
import { useProduct } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProduct';
import Price from '@magento/venia-ui/lib/components/Price';

import { useStyle } from '@magento/venia-ui/lib/classify';
import Image from '@magento/venia-ui/lib/components/Image';
import Kebab from '@magento/venia-ui/lib/components/LegacyMiniCart/kebab';
import ProductOptions from '@magento/venia-ui/lib/components/LegacyMiniCart/productOptions';
import Section from '@magento/venia-ui/lib/components/LegacyMiniCart/section';
import Quantity from '@magento/venia-ui/lib/components/CartPage/ProductListing/quantity';

import defaultClasses from '@magento/venia-ui/lib/components/CartPage/ProductListing/product.css';

import { CartPageFragment } from '@magento/venia-ui/lib/components/CartPage/cartPageFragments.gql';
import { AvailableShippingMethodsCartFragment } from '@magento/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingMethodsFragments.gql';

const IMAGE_SIZE = 100;

const Product = props => {
    const { item } = props;

    const { formatMessage } = useIntl();
    const talonProps = useProduct({
        operations: {
            removeItemMutation: REMOVE_ITEM_MUTATION,
            updateItemQuantityMutation: UPDATE_QUANTITY_MUTATION
        },
        ...props,
        item: {
            ...item,
            id: parseInt(item.id)
        }
    });

    const {
        errorMessage,
        handleEditItem,
        handleRemoveFromCart,
        handleUpdateItemQuantity,
        isEditable,
        product,
        isProductUpdating
    } = talonProps;

    const { currency, image, name, options, quantity, stockStatus, unitPrice } = product;

    const classes = useStyle(defaultClasses, props.classes);

    const itemClassName = isProductUpdating ? classes.item_disabled : classes.item;

    const editItemSection = isEditable ? (
        <Section
            text={formatMessage({
                id: 'product.editItem',
                defaultMessage: 'Edit item'
            })}
            onClick={handleEditItem}
            icon="Edit2"
            classes={{
                text: classes.sectionText
            }}
        />
    ) : null;

    const stockStatusMessage =
        stockStatus === 'OUT_OF_STOCK'
            ? formatMessage({
                  id: 'product.outOfStock',
                  defaultMessage: 'Out-of-stock'
              })
            : '';

    return (
        <li className={classes.root}>
            <span className={classes.errorText}>{errorMessage}</span>
            <div className={itemClassName}>
                <div className={classes.imageContainer}>
                    <Image
                        alt={name}
                        classes={{
                            root: classes.imageRoot,
                            image: classes.image
                        }}
                        width={IMAGE_SIZE}
                        resource={image}
                    />
                </div>
                <div className={classes.details}>
                    <div className={classes.name}>{name}</div>
                    <ProductOptions
                        options={options}
                        classes={{
                            options: classes.options,
                            optionLabel: classes.optionLabel
                        }}
                    />
                    <span className={classes.price}>
                        <Price currencyCode={currency} value={unitPrice} />
                        <FormattedMessage id={'product.price'} defaultMessage={' ea.'} />
                    </span>
                    <span className={classes.stockStatusMessage}>{stockStatusMessage}</span>
                    <div className={classes.quantity}>
                        <Quantity itemId={item.id} initialValue={quantity} onChange={handleUpdateItemQuantity} />
                    </div>
                </div>
                <Kebab
                    classes={{
                        root: classes.kebab
                    }}
                    disabled={true}
                >
                    {editItemSection}
                    <Section
                        text={formatMessage({
                            id: 'product.removeFromCart',
                            defaultMessage: 'Remove from cart'
                        })}
                        onClick={handleRemoveFromCart}
                        icon="Trash"
                        classes={{
                            text: classes.sectionText
                        }}
                    />
                </Kebab>
            </div>
        </li>
    );
};

export default Product;

export const REMOVE_ITEM_MUTATION = gql`
    mutation removeItem($cartId: String!, $itemId: Int!) {
        removeItemFromCart(input: { cart_id: $cartId, cart_item_id: $itemId }) @connection(key: "removeItemFromCart") {
            cart {
                id
                ...CartPageFragment
                ...AvailableShippingMethodsCartFragment
            }
        }
    }
    ${CartPageFragment}
    ${AvailableShippingMethodsCartFragment}
`;

export const UPDATE_QUANTITY_MUTATION = gql`
    mutation updateItemQuantity($cartId: String!, $itemId: Int!, $quantity: Float!) {
        updateCartItems(input: { cart_id: $cartId, cart_items: [{ cart_item_id: $itemId, quantity: $quantity }] })
            @connection(key: "updateCartItems") {
            cart {
                id
                ...CartPageFragment
                ...AvailableShippingMethodsCartFragment
            }
        }
    }
    ${CartPageFragment}
    ${AvailableShippingMethodsCartFragment}
`;
