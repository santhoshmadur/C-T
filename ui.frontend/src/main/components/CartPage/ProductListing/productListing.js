/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
import React, { Fragment, Suspense } from 'react';
import { FormattedMessage } from 'react-intl';
import { gql } from '@apollo/client';
import { useProductListing } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProductListing';

import { useStyle } from '@magento/venia-ui/lib/classify';
import LoadingIndicator from '@magento/venia-ui/lib/components/LoadingIndicator';
import defaultClasses from '@magento/venia-ui/lib/components/CartPage/ProductListing/productListing.css';
import Product from './product';
import { ProductListingFragment } from '@magento/venia-ui/lib/components/CartPage/ProductListing/productListingFragments';

const EditModal = React.lazy(
    async () => await import('@magento/venia-ui/lib/components/CartPage/ProductListing/EditModal').default
);
/**
 * A child component of the CartPage component.
 * This component renders the product listing on the cart page.
 *
 * @param {Object} props
 * @param {Function} props.setIsCartUpdating Function for setting the updating state of the cart.
 * @param {Object} props.classes CSS className overrides.
 * See [productListing.css]{@link https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/ProductListing/productListing.css}
 * for a list of classes you can override.
 *
 * @returns {React.Element}
 *
 * @example <caption>Importing into your project</caption>
 * import ProductListing from "@magento/venia-ui/lib/components/CartPage/ProductListing";
 */
const ProductListing = props => {
    const { onAddToWishlistSuccess, setIsCartUpdating, fetchCartDetails } = props;
    const talonProps = useProductListing({
        queries: {
            getProductListing: GET_PRODUCT_LISTING
        }
    });
    const { activeEditItem, isLoading, items, setActiveEditItem, wishlistConfig } = talonProps;

    const classes = useStyle(defaultClasses, props.classes);

    if (isLoading) {
        return (
            <LoadingIndicator>
                <FormattedMessage id={'productListing.loading'} defaultMessage={'Fetching Cart...'} />
            </LoadingIndicator>
        );
    }

    if (items.length) {
        const productComponents = items.map(product => (
            <Product
                item={product}
                key={product.id}
                setActiveEditItem={setActiveEditItem}
                setIsCartUpdating={setIsCartUpdating}
                onAddToWishlistSuccess={onAddToWishlistSuccess}
                fetchCartDetails={fetchCartDetails}
                wishlistConfig={wishlistConfig}
            />
        ));

        return (
            <Fragment>
                <ul className={classes.root}>{productComponents}</ul>
                <Suspense fallback={null}>
                    <EditModal
                        item={activeEditItem}
                        setIsCartUpdating={setIsCartUpdating}
                        setActiveEditItem={setActiveEditItem}
                    />
                </Suspense>
            </Fragment>
        );
    }

    return null;
};

export const GET_PRODUCT_LISTING = gql`
    query getProductListing($cartId: String!) {
        cart(cart_id: $cartId) {
            id
            ...ProductListingFragment
        }
    }
    ${ProductListingFragment}
`;

export default ProductListing;
