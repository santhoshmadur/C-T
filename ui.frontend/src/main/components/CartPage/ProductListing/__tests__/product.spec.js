/*******************************************************************************
 *
 *    Copyright 2021 Adobe. All rights reserved.
 *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License. You may obtain a copy
 *    of the License at http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software distributed under
 *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
 *    OF ANY KIND, either express or implied. See the License for the specific language
 *    governing permissions and limitations under the License.
 *
 ******************************************************************************/
jest.mock('@magento/venia-ui/lib/components/Image', () => 'Image');
jest.mock('@magento/peregrine/lib/talons/CartPage/ProductListing/useProduct');
jest.mock('@apollo/client', () => {
    const executeMutation = jest.fn(() => ({ error: null }));
    const useMutation = jest.fn(() => [executeMutation]);

    return {
        gql: jest.fn(),
        useMutation
    };
});

jest.mock('@magento/peregrine/lib/context/cart', () => {
    const state = { cartId: 'cart123' };
    const api = {};
    const useCartContext = jest.fn(() => [state, api]);

    return { useCartContext };
});

jest.mock('react-router-dom', () => ({
    Link: ({ children, ...rest }) => <div {...rest}>{children}</div> // eslint-disable-line react/display-name
}));
jest.mock('@magento/peregrine/lib/util/makeUrl');

import React from 'react';
import { useProduct } from '@magento/peregrine/lib/talons/CartPage/ProductListing/useProduct';

import Product from '../product';
import render from '../../../utils/test-utils';

const props = {
    item: {
        id: '123',
        product: {
            name: 'Unit Test Product',
            small_image: {
                url: 'unittest.jpg'
            },
            urlKey: 'unittest',
            urlSuffix: '.html',
            sku: '12345'
        },
        prices: {
            price: {
                currency: 'USD',
                value: 100
            }
        },
        quantity: 1,
        configurable_options: [
            {
                configurable_product_option_value_uid: '12345asd'
            },
            {
                configurable_product_option_value_uid: 'asf2134'
            }
        ]
    }
};

describe('product', () => {
    test('renders simple product correctly', () => {
        useProduct.mockReturnValueOnce({
            addToWishlistProps: {
                atwProp1: 'value1'
            },
            errorMessage: undefined,
            handleEditItem: jest.fn(),
            handleRemoveFromCart: jest.fn(),
            handleUpdateItemQuantity: jest.fn(),
            isEditable: false,
            product: {
                currency: 'USD',
                image: {},
                name: '',
                options: [],
                quantity: 1,
                unitPrice: 1,
                urlKey: 'unittest',
                urlSuffix: '.html'
            },
            isProductUpdating: false
        });
        const tree = render(<Product {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('renders out of stock product', () => {
        useProduct.mockReturnValueOnce({
            addToWishlistProps: {
                atwProp1: 'value1'
            },
            errorMessage: undefined,
            handleEditItem: jest.fn(),
            handleRemoveFromCart: jest.fn(),
            handleUpdateItemQuantity: jest.fn(),
            isEditable: false,
            product: {
                currency: 'USD',
                image: {},
                name: '',
                options: [],
                quantity: 2,
                stockStatus: 'OUT_OF_STOCK',
                unitPrice: 55,
                urlKey: 'popular-product',
                urlSuffix: ''
            },
            loginToastProps: null,
            isProductUpdating: false
        });
        const tree = render(<Product {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });

    test('renders configurable product with options', () => {
        useProduct.mockReturnValueOnce({
            addToWishlistProps: {
                atwProp1: 'value1'
            },
            errorMessage: undefined,
            handleEditItem: jest.fn(),
            handleRemoveFromCart: jest.fn(),
            handleUpdateItemQuantity: jest.fn(),
            isEditable: true,
            product: {
                currency: 'USD',
                image: {},
                name: '',
                urlKey: 'unittest',
                urlSuffix: '.html',
                options: [
                    {
                        option_label: 'Option 1',
                        value_label: 'Value 1'
                    },
                    {
                        option_label: 'Option 2',
                        value_label: 'Value 2'
                    }
                ],
                quantity: 1,
                unitPrice: 1
            },
            loginToastProps: null,
            isProductUpdating: false
        });

        const tree = render(<Product {...props} />);

        expect(tree.toJSON()).toMatchSnapshot();
    });
});
