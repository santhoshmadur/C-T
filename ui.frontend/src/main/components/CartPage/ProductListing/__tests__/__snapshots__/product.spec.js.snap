// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product renders configurable product with options 1`] = `
<li>
  <span />
  <div>
    <div>
      <Image
        alt=""
        classes={
          Object {
            "image": undefined,
            "root": undefined,
          }
        }
        resource={Object {}}
        width={100}
      />
    </div>
    <div>
      <div>
        
      </div>
      <dl>
        <div>
          <dt>
            Option 1 :
          </dt>
          <dd>
            Value 1
          </dd>
        </div>
        <div>
          <dt>
            Option 2 :
          </dt>
          <dd>
            Value 2
          </dd>
        </div>
      </dl>
      <span>
        <span>
          $
        </span>
        <span>
          1
        </span>
        <span>
          .
        </span>
        <span>
          00
        </span>
         ea.
      </span>
      <span>
        
      </span>
      <div>
        <form
          onKeyDown={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div>
            <label
              htmlFor="123"
            >
              Quantity
            </label>
            <button
              aria-label="Decrease Quantity"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <span>
                <svg
                  fill="none"
                  height={22}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={22}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
            <span
              style={
                Object {
                  "--iconsAfter": 0,
                  "--iconsBefore": 0,
                }
              }
            >
              <span>
                <input
                  aria-label="Item Quantity"
                  id="123"
                  inputMode="numeric"
                  min={0}
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  pattern="[0-9]*"
                  value={1}
                />
              </span>
              <span />
              <span />
            </span>
            <p />
            <button
              aria-label="Increase Quantity"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                <svg
                  fill="none"
                  height={20}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <span>
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </span>
      </button>
      <ul>
        <li>
          <button
            onMouseDown={[MockFunction]}
          >
            <span>
              <svg
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M17 3a2.828 2.828 0 1 1 4 4L7.5 20.5 2 22l1.5-5.5L17 3z"
                />
              </svg>
            </span>
            <span>
              Edit item
            </span>
          </button>
        </li>
        <li>
          <button
            onMouseDown={[MockFunction]}
          >
            <span>
              <svg
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="3 6 5 6 21 6"
                />
                <path
                  d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                />
              </svg>
            </span>
            <span>
              Remove from cart
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</li>
`;

exports[`product renders out of stock product 1`] = `
<li>
  <span />
  <div>
    <div>
      <Image
        alt=""
        classes={
          Object {
            "image": undefined,
            "root": undefined,
          }
        }
        resource={Object {}}
        width={100}
      />
    </div>
    <div>
      <div>
        
      </div>
      <span>
        <span>
          $
        </span>
        <span>
          55
        </span>
        <span>
          .
        </span>
        <span>
          00
        </span>
         ea.
      </span>
      <span>
        Out-of-stock
      </span>
      <div>
        <form
          onKeyDown={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div>
            <label
              htmlFor="123"
            >
              Quantity
            </label>
            <button
              aria-label="Decrease Quantity"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                <svg
                  fill="none"
                  height={22}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={22}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
            <span
              style={
                Object {
                  "--iconsAfter": 0,
                  "--iconsBefore": 0,
                }
              }
            >
              <span>
                <input
                  aria-label="Item Quantity"
                  id="123"
                  inputMode="numeric"
                  min={0}
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  pattern="[0-9]*"
                  value={2}
                />
              </span>
              <span />
              <span />
            </span>
            <p />
            <button
              aria-label="Increase Quantity"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                <svg
                  fill="none"
                  height={20}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <span>
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </span>
      </button>
      <ul>
        <li>
          <button
            onMouseDown={[MockFunction]}
          >
            <span>
              <svg
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="3 6 5 6 21 6"
                />
                <path
                  d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                />
              </svg>
            </span>
            <span>
              Remove from cart
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</li>
`;

exports[`product renders simple product correctly 1`] = `
<li>
  <span />
  <div>
    <div>
      <Image
        alt=""
        classes={
          Object {
            "image": undefined,
            "root": undefined,
          }
        }
        resource={Object {}}
        width={100}
      />
    </div>
    <div>
      <div>
        
      </div>
      <span>
        <span>
          $
        </span>
        <span>
          1
        </span>
        <span>
          .
        </span>
        <span>
          00
        </span>
         ea.
      </span>
      <span>
        
      </span>
      <div>
        <form
          onKeyDown={[Function]}
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div>
            <label
              htmlFor="123"
            >
              Quantity
            </label>
            <button
              aria-label="Decrease Quantity"
              disabled={true}
              onClick={[Function]}
              type="button"
            >
              <span>
                <svg
                  fill="none"
                  height={22}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={22}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
            <span
              style={
                Object {
                  "--iconsAfter": 0,
                  "--iconsBefore": 0,
                }
              }
            >
              <span>
                <input
                  aria-label="Item Quantity"
                  id="123"
                  inputMode="numeric"
                  min={0}
                  name="quantity"
                  onBlur={[Function]}
                  onChange={[Function]}
                  pattern="[0-9]*"
                  value={1}
                />
              </span>
              <span />
              <span />
            </span>
            <p />
            <button
              aria-label="Increase Quantity"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <span>
                <svg
                  fill="none"
                  height={20}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={20}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="12"
                    x2="12"
                    y1="5"
                    y2="19"
                  />
                  <line
                    x1="5"
                    x2="19"
                    y1="12"
                    y2="12"
                  />
                </svg>
              </span>
            </button>
          </div>
        </form>
      </div>
    </div>
    <div>
      <button
        onClick={[Function]}
      >
        <span>
          <svg
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="12"
              cy="5"
              r="1"
            />
            <circle
              cx="12"
              cy="19"
              r="1"
            />
          </svg>
        </span>
      </button>
      <ul>
        <li>
          <button
            onMouseDown={[MockFunction]}
          >
            <span>
              <svg
                fill="none"
                height={16}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="3 6 5 6 21 6"
                />
                <path
                  d="M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2"
                />
              </svg>
            </span>
            <span>
              Remove from cart
            </span>
          </button>
        </li>
      </ul>
    </div>
  </div>
</li>
`;
